File: Sample/ContextApi.CS/ContextApi.Tests/UnitTestsBase.cs
using ContextApi.Tests;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Telerik.OpenAccess;

namespace ContextApi.Tests
{
    [TestClass]
    public abstract class UnitTestsBase
    {
        [TestInitialize]
        public virtual void TestInitialize()
        {
            LocalDbInstanceManager.CreateInstance();
            ContextOperations.ClearAllEntites();
            ContextOperations.AddEntities();
        }

        [TestCleanup]
        public virtual void TestCleanup()
        {
            ContextOperations.ClearAllEntites();
        }

        //Get the object state of a persistent object.
        protected ObjectState GetObjectState(object entity)
        {
            return OpenAccessContext.PersistenceState.GetState(entity);
        }
    }
}

File: Sample/ContextApi.VB/ContextApi.Tests/UnitTestsBase.vb
Imports Telerik.OpenAccess
Imports Microsoft.VisualStudio.TestTools.UnitTesting

<TestClass> _
Public Class UnitTestsBase

    <TestInitialize> _
    Public Overridable Sub TestInitialize()
        LocalDbInstanceManager.CreateInstance()
        ContextOperations.ClearAllEntites()
        ContextOperations.AddEntities()
    End Sub

    <TestCleanup> _
    Public Overridable Sub TestCleanup()
        ContextOperations.ClearAllEntites()
    End Sub

    'Get the object state of a persistent object detached from the context.
    Protected Function GetObjectState(entity As Object) As ObjectState
        Return OpenAccessContext.PersistenceState.GetState(entity)
    End Function

End Class
